source/server/config_validation/server.cc

```diff
@@ -0,0 +1,65 @@
+#include "server/config_validation/server.h"
+
+#include "common/json/config_schemas.h"
+
+#include "server/configuration_impl.h"
+
+namespace Envoy {
+namespace Server {
+
+ValidationInstance::ValidationInstance(Options& options, HotRestart& restarter,
+                                       Stats::IsolatedStoreImpl& store,
+                                       Thread::BasicLockable& access_log_lock,
+                                       ComponentFactory& component_factory,
+                                       const LocalInfo::LocalInfo& local_info)
+    : options_(options), restarter_(restarter), stats_store_(store),
+      handler_(Api::ApiPtr{new Api::ValidationImpl(options.fileFlushIntervalMsec())}),
+      local_info_(local_info),
+      access_log_manager_(handler_.api(), handler_.dispatcher(), access_log_lock, store) {
+  try {
+    initialize(options, component_factory);
+  } catch (const EnvoyException& e) {
+    log().critical("error initializing configuration '{}': {}", options.configPath(), e.what());
+    thread_local_.shutdownThread();
+    exit(1);
+  }
+}
+
+void ValidationInstance::initialize(Options& options, ComponentFactory& component_factory) {
+  // See comments on InstanceImpl::initialize() for the overall flow here.
+  //
+  // For validation, we only do a subset of normal server initialization: everything that could fail
+  // on a malformed config (e.g. JSON parsing and all the object construction that follows), but
+  // more importantly nothing with observable effects (e.g. binding to ports or shutting down any
+  // other Envoy process).
+  //
+  // If we get all the way through that stripped-down initialization flow, to the point where we'd
+  // be ready to serve, then the config has passed validation.
+  Json::ObjectPtr config_json = Json::Factory::loadFromFile(options.configPath());
+  config_json->validateSchema(Json::Schema::TOP_LEVEL_CONFIG_SCHEMA);
```

Comment:
I think this is just a bug. @rlazarus can you move this validation into the Configuration::InitialImpl constructor where it should be, then it can be removed from the prod server also.